kiwmi/desktop/layer_shell.c

```diff
@@ -24,12 +24,14 @@ kiwmi_layer_destroy_notify(struct wl_listener *listener, void *UNUSED(data))
 {
     struct kiwmi_layer *layer = wl_container_of(listener, layer, destroy);
 
-    wl_list_remove(&layer->link);
     wl_list_remove(&layer->destroy.link);
     wl_list_remove(&layer->map.link);
     wl_list_remove(&layer->unmap.link);
 
-    arrange_layers(layer->output);
+    if (NULL != layer->output) {
```

Comment:
I believe the correct solution would be to destroy all layers in output destroy handler. I even think i’ve got that somewhere in #50, feel free to copy it if you wish.

And a style nitpick (also elsewhere)
```suggestion
    if (layer->output != NULL) {
```

Reply:
Duly copied, many thanks.

And I've switched my conditions around, MISRA habits come back to bite me.